modules/dxp/apps/segments/segments-experiment-web/src/main/java/com/liferay/segments/experiment/web/internal/processor/SegmentsExperimentSegmentsExperienceRequestProcessor.java

```diff
@@ -300,9 +305,14 @@ private long _getSelectedSegmentsExperienceId(
 				_segmentsExperienceLocalService.fetchSegmentsExperience(
 					selectedSegmentsExperienceId);
 
-			if (segmentsExperience != null) {
```

Comment:
Could you add a new test in SegmentsExperimentSegmentsExperienceRequestProcessorTest?

that test the case when themeDisplay.getPlid() != segmentsExperience.getClassPK() the default experience is returned?